File: example/systems/HealthRenderSystem.java
package net.brokenspork.systems;

import net.brokenspork.components.Health;
import net.brokenspork.components.Position;

import com.artemis.Aspect;
import com.artemis.ComponentMapper;
import com.artemis.Entity;
import com.artemis.annotations.Mapper;
import com.artemis.systems.EntityProcessingSystem;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.OrthographicCamera;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.Texture.TextureFilter;
import com.badlogic.gdx.graphics.g2d.BitmapFont;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.graphics.g2d.TextureRegion;
import com.badlogic.gdx.math.MathUtils;

public class HealthRenderSystem extends EntityProcessingSystem {
	@Mapper ComponentMapper<Position> pm;
	@Mapper ComponentMapper<Health> hm;
	
	private SpriteBatch batch;
	private OrthographicCamera camera;
	private BitmapFont font;
	
	@SuppressWarnings("unchecked")
    public HealthRenderSystem(OrthographicCamera camera) {
		super(Aspect.getAspectForAll(Position.class, Health.class));
		this.camera = camera;
	}
	
	@Override
	protected void initialize() {
		batch = new SpriteBatch();
		
		Texture fontTexture = new Texture(Gdx.files.internal("fonts/normal_0.png"));
		fontTexture.setFilter(TextureFilter.Linear, TextureFilter.MipMapLinearLinear);
		TextureRegion fontRegion = new TextureRegion(fontTexture);
		font = new BitmapFont(Gdx.files.internal("fonts/normal.fnt"), fontRegion, false);
		font.setUseIntegerPositions(false);
	}
	
	@Override
	protected void begin() {
		batch.setProjectionMatrix(camera.combined);
		batch.begin();
	}

	@Override
	protected void process(Entity e) {
		Position position = pm.get(e);
		Health health = hm.get(e);
		
		int percentage = MathUtils.round(health.health/health.maximumHealth*100f);
		
		font.draw(batch, percentage+"%", position.x, position.y);
	}
	
	@Override
	protected void end() {
		batch.end();
	}
}

File: example/systems/HealthRenderSystem.ts
module brokenspork.systems {
	
	import Health = brokenspork.components.Health;
	import Position = brokenspork.components.Position;
  import Sprite = brokenspork.components.Sprite;

	import Aspect = artemis.Aspect;
	import ComponentMapper = artemis.ComponentMapper;
	import Entity = artemis.Entity;
	import EntityProcessingSystem = artemis.systems.EntityProcessingSystem;
	import MathUtils = artemis.utils.MathUtils;
	import Mapper = artemis.annotations.Mapper;

	export class HealthRenderSystem extends EntityProcessingSystem {
		@Mapper(Position) pm:ComponentMapper<Position>;
		@Mapper(Health) hm:ComponentMapper<Health>;
		
		//private batch:SpriteBatch;
		// private OrthographicCamera camera;
		// private BitmapFont font;
		
		public constructor() {
			super(Aspect.getAspectForAll(Position, Health));
		}
		
		
		public initialize() {
			// batch = new SpriteBatch();
			
			// Texture fontTexture = new Texture(Gdx.files.internal("fonts/normal_0.png"));
			// fontTexture.setFilter(TextureFilter.Linear, TextureFilter.MipMapLinearLinear);
			// TextureRegion fontRegion = new TextureRegion(fontTexture);
			// font = new BitmapFont(Gdx.files.internal("fonts/normal.fnt"), fontRegion, false);
			// font.setUseIntegerPositions(false);
		}
		
		
		protected begin() {
			// batch.setProjectionMatrix(camera.combined);
			// batch.begin();
		}

    //public inserted(e:Entity) {
    //  var c:Sprite = e.getComponentByType(Sprite);
    //  console.log('HealthRenderSystem::inserted', c.name, e.uuid);
    //}
    //protected removed(e:Entity) {
    //  var c:Sprite = e.getComponentByType(Sprite);
    //  console.log('HealthRenderSystem::removed', c.name, e.uuid);
    //}

		public processEach(e:Entity) {
			var position:Position = this.pm.get(e);
			var health:Health = this.hm.get(e);
			
			var percentage:number = Math.round(health.health/health.maximumHealth*100);
			//font.draw(batch, percentage+"%", position.x, position.y);
		}


		protected end() {
			//batch.end();
		}
	}
}
